tests/golos.ctrl_tests.cpp

```diff
@@ -458,5 +460,53 @@ BOOST_FIXTURE_TEST_CASE(update_auths, golos_ctrl_tester) try {
     BOOST_CHECK(top_witnesses["witnesses"].as<std::vector<name>>().size() == wtns.size());
 } FC_LOG_AND_RETHROW()
 
+BOOST_FIXTURE_TEST_CASE(payments_top_witness, golos_ctrl_tester) try {
+    prepare_balances();
+    BOOST_CHECK_EQUAL(success(), token.open(cfg::control_name, _token, cfg::control_name));
+    BOOST_CHECK_EQUAL(success(), token.issue(_issuer, _issuer, dasset(10000), "on emission"));
+
+    BOOST_CHECK_EQUAL(err.not_valid_ctrl_params, token.transfer(_issuer, cfg::control_name, dasset(10000), "emission"));
+
+    BOOST_CHECK_EQUAL(success(), token.create(_issuer, asset(100500, symbol(4, "GOLOS"))));
+    BOOST_CHECK_EQUAL(success(), token.issue(_issuer, _issuer, asset(100000, symbol(4, "GOLOS")), "issue"));
+    BOOST_CHECK_EQUAL(success(), token.open(cfg::control_name, symbol(4, "GOLOS"), cfg::control_name));
+
+    BOOST_CHECK_EQUAL(success(), ctrl.set_params(_test_params));
+    produce_block();
+    ctrl.prepare_multisig(BLOG);
+    produce_block();
+
+    BOOST_CHECK_EQUAL(err.not_valid_ctrl_params, token.transfer(_issuer, cfg::control_name, asset(1000, symbol(4, "GOLOS")), "emission"));
+    BOOST_CHECK_EQUAL(success(), token.transfer(_issuer, cfg::control_name, dasset(1000), "emission"));
+    BOOST_CHECK_EQUAL(token.get_account(cfg::control_name)["balance"].as<asset>(), token.make_asset(1000));
+
+    for (int i = 0; i < _n_w; i++)
+        BOOST_CHECK_EQUAL(success(), ctrl.reg_witness(_w[i], "localhost"));
+    produce_block();
+
+    vector<std::pair<name,name>> votes = {
+        {_alice, _w[0]}, {_alice, _w[1]}, {_alice, _w[2]}, {_alice, _w[3]},
+        {_bob, _w[0]}, {_w[0], _w[0]}
+    };
+    for (const auto& v : votes)
+        BOOST_CHECK_EQUAL(success(), ctrl.vote_witness(v.first, v.second));
+    produce_block();
+
+    BOOST_CHECK_EQUAL(success(), token.transfer(_issuer, cfg::control_name, dasset(1000), "emission"));
+    BOOST_CHECK_EQUAL(token.get_account(cfg::control_name)["balance"].as<asset>(), token.make_asset(1000));
+
+    BOOST_CHECK_EQUAL(token.get_account(_w[0])["payments"].as<asset>(), token.make_asset(500));
+    BOOST_CHECK_EQUAL(token.get_account(_w[1])["payments"].as<asset>(), token.make_asset(500));
+    BOOST_CHECK_EQUAL(token.get_account(_w[2])["payments"].as<asset>(), token.make_asset(0));
+    BOOST_CHECK_EQUAL(token.get_account(_w[3])["payments"].as<asset>(), token.make_asset(0));
+    BOOST_CHECK_EQUAL(token.get_account(_w[4])["payments"].as<asset>(), token.make_asset(0));
+
+    // need tapos_block_prefix to choose a winner
+    BOOST_CHECK_EQUAL(success(), token.transfer(_issuer, cfg::control_name, dasset(3.333), "emission"));
+    BOOST_CHECK_EQUAL(token.get_account(_w[0])["payments"].as<asset>(), token.make_asset(501.666));
+    BOOST_CHECK_EQUAL(token.get_account(_w[1])["payments"].as<asset>(), token.make_asset(501.667));
```

Comment:
Please add test of on_transfer when raw amount is less than number of recipients e.g. there are `10` witnesses, and amount is `0.009`, so all witnesses get 0 and "winner" gets all